# TP1

In [ ]:
import pandas as pd

## Charger vos données dans un DataFrame Pandas

In [ ]:
data = pd.read_excel('titanic.xls')

## On indexe les données par nom

In [ ]:
data1 = data.set_index('name')
data1

pclass  survived     sex  \
name                                                                        
Allen, Miss. Elisabeth Walton                         1         1  female   
Allison, Master. Hudson Trevor                        1         1    male   
Allison, Miss. Helen Loraine                          1         0  female   
Allison, Mr. Hudson Joshua Creighton                  1         0    male   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)       1         0  female   
...                                                 ...       ...     ...   
Zabour, Miss. Hileni                                  3         0  female   
Zabour, Miss. Thamine                                 3         0  female   
Zakarian, Mr. Mapriededer                             3         0    male   
Zakarian, Mr. Ortin                                   3         0    male   
Zimmerman, Mr. Leo                                    3         0    male   

                                                     age  sibsp  parch  \
name                                                                     
Allen, Miss. Elisabeth Walton                    29.0000      0      0   
Allison, Master. Hudson Trevor                    0.9167      1      2   
Allison, Miss. Helen Loraine                      2.0000      1      2   
Allison, Mr. Hudson Joshua Creighton             30.0000      1      2   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  25.0000      1      2   
...                                                  ...    ...    ...   
Zabour, Miss. Hileni                             14.5000      1      0   
Zabour, Miss. Thamine                                NaN      1      0   
Zakarian, Mr. Mapriededer                        26.5000      0      0   
Zakarian, Mr. Ortin                              27.0000      0      0   
Zimmerman, Mr. Leo                               29.0000      0      0   

                                                 ticket      fare    cabin  \
name                                                                         
Allen, Miss. Elisabeth Walton                     24160  211.3375       B5   
Allison, Master. Hudson Trevor                   113781  151.5500  C22 C26   
Allison, Miss. Helen Loraine                     113781  151.5500  C22 C26   
Allison, Mr. Hudson Joshua Creighton             113781  151.5500  C22 C26   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  113781  151.5500  C22 C26   
...                                                 ...       ...      ...   
Zabour, Miss. Hileni                               2665   14.4542      NaN   
Zabour, Miss. Thamine                              2665   14.4542      NaN   
Zakarian, Mr. Mapriededer                          2656    7.2250      NaN   
Zakarian, Mr. Ortin                                2670    7.2250      NaN   
Zimmerman, Mr. Leo                               315082    7.8750      NaN   

                                                embarked boat   body  \
name                                                                   
Allen, Miss. Elisabeth Walton                          S    2    NaN   
Allison, Master. Hudson Trevor                         S   11    NaN   
Allison, Miss. Helen Loraine                           S  NaN    NaN   
Allison, Mr. Hudson Joshua Creighton                   S  NaN  135.0   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)        S  NaN    NaN   
...                                                  ...  ...    ...   
Zabour, Miss. Hileni                                   C  NaN  328.0   
Zabour, Miss. Thamine                                  C  NaN    NaN   
Zakarian, Mr. Mapriededer                              C  NaN  304.0   
Zakarian, Mr. Ortin                                    C  NaN    NaN   
Zimmerman, Mr. Leo                                     S  NaN    NaN   

                                                                       home.dest  
name                                             

## Affichez la dimension, les 7 premiers éléments

In [ ]:
#Dimension
data1.shape

(1309, 13)

In [ ]:
#Les 7 premiers éléments
data1.head(7)

pclass  survived     sex  \
name                                                                        
Allen, Miss. Elisabeth Walton                         1         1  female   
Allison, Master. Hudson Trevor                        1         1    male   
Allison, Miss. Helen Loraine                          1         0  female   
Allison, Mr. Hudson Joshua Creighton                  1         0    male   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)       1         0  female   
Anderson, Mr. Harry                                   1         1    male   
Andrews, Miss. Kornelia Theodosia                     1         1  female   

                                                     age  sibsp  parch  \
name                                                                     
Allen, Miss. Elisabeth Walton                    29.0000      0      0   
Allison, Master. Hudson Trevor                    0.9167      1      2   
Allison, Miss. Helen Loraine                      2.0000      1      2   
Allison, Mr. Hudson Joshua Creighton             30.0000      1      2   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  25.0000      1      2   
Anderson, Mr. Harry                              48.0000      0      0   
Andrews, Miss. Kornelia Theodosia                63.0000      1      0   

                                                 ticket      fare    cabin  \
name                                                                         
Allen, Miss. Elisabeth Walton                     24160  211.3375       B5   
Allison, Master. Hudson Trevor                   113781  151.5500  C22 C26   
Allison, Miss. Helen Loraine                     113781  151.5500  C22 C26   
Allison, Mr. Hudson Joshua Creighton             113781  151.5500  C22 C26   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  113781  151.5500  C22 C26   
Anderson, Mr. Harry                               19952   26.5500      E12   
Andrews, Miss. Kornelia Theodosia                 13502   77.9583       D7   

                                                embarked boat   body  \
name                                                                   
Allen, Miss. Elisabeth Walton                          S    2    NaN   
Allison, Master. Hudson Trevor                         S   11    NaN   
Allison, Miss. Helen Loraine                           S  NaN    NaN   
Allison, Mr. Hudson Joshua Creighton                   S  NaN  135.0   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)        S  NaN    NaN   
Anderson, Mr. Harry                                    S    3    NaN   
Andrews, Miss. Kornelia Theodosia                      S   10    NaN   

                                                                       home.dest  
name                                                                              
Allen, Miss. Elisabeth Walton                                       St Louis, MO  
Allison, Master. Hudson Trevor                   Montreal, PQ / Chesterville, ON  
Allison, Miss. Helen Loraine                     Montreal, PQ / Chesterville, ON  
Allison, Mr. Hudson Joshua Creighton             Montreal, PQ / Chesterville, ON  
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  Montreal, PQ / Chesterville, ON  
Anderson, Mr. Harry                                                 New York, NY  
Andrews, Miss. Kornelia Theodosia                                     Hudson, NY

## Affichez les indicateurs statistiques

In [ ]:
data1.describe()

pclass     survived          age        sibsp        parch  \
count  1309.000000  1309.000000  1046.000000  1309.000000  1309.000000   
mean      2.294882     0.381971    29.881135     0.498854     0.385027   
std       0.837836     0.486055    14.413500     1.041658     0.865560   
min       1.000000     0.000000     0.166700     0.000000     0.000000   
25%       2.000000     0.000000    21.000000     0.000000     0.000000   
50%       3.000000     0.000000    28.000000     0.000000     0.000000   
75%       3.000000     1.000000    39.000000     1.000000     0.000000   
max       3.000000     1.000000    80.000000     8.000000     9.000000   

              fare        body  
count  1308.000000  121.000000  
mean     33.295479  160.809917  
std      51.758668   97.696922  
min       0.000000    1.000000  
25%       7.895800   72.000000  
50%      14.454200  155.000000  
75%      31.275000  256.000000  
max     512.329200  328.000000

## Nettoyer votre Dataset avec drop(), dropna()
### Gardez uniquement les colonnes <code>pclass	survived	sex	age</code>

In [ ]:
data1=data1.drop(['sibsp', 'parch','ticket', 'fare', 'cabin', 'embarked', 'boat', 'body', 'home.dest'], axis=1)
data1

pclass  survived     sex  \
name                                                                        
Allen, Miss. Elisabeth Walton                         1         1  female   
Allison, Master. Hudson Trevor                        1         1    male   
Allison, Miss. Helen Loraine                          1         0  female   
Allison, Mr. Hudson Joshua Creighton                  1         0    male   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)       1         0  female   
...                                                 ...       ...     ...   
Zabour, Miss. Hileni                                  3         0  female   
Zabour, Miss. Thamine                                 3         0  female   
Zakarian, Mr. Mapriededer                             3         0    male   
Zakarian, Mr. Ortin                                   3         0    male   
Zimmerman, Mr. Leo                                    3         0    male   

                                                     age  
name                                                      
Allen, Miss. Elisabeth Walton                    29.0000  
Allison, Master. Hudson Trevor                    0.9167  
Allison, Miss. Helen Loraine                      2.0000  
Allison, Mr. Hudson Joshua Creighton             30.0000  
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  25.0000  
...                                                  ...  
Zabour, Miss. Hileni                             14.5000  
Zabour, Miss. Thamine                                NaN  
Zakarian, Mr. Mapriededer                        26.5000  
Zakarian, Mr. Ortin                              27.0000  
Zimmerman, Mr. Leo                               29.0000  

[1309 rows x 4 columns]

## Affichez les indicateurs statistiques et le nombre de valeurs manquantes

In [ ]:
#Les indicateurs statistiques
data1.describe()

pclass     survived          age
count  1309.000000  1309.000000  1046.000000
mean      2.294882     0.381971    29.881135
std       0.837836     0.486055    14.413500
min       1.000000     0.000000     0.166700
25%       2.000000     0.000000    21.000000
50%       3.000000     0.000000    28.000000
75%       3.000000     1.000000    39.000000
max       3.000000     1.000000    80.000000

In [ ]:
#Le nombre de valeurs manquantes
print("Le nombre de valeurs manquantes est :", data1.isnull().sum().sum())

Le nombre de valeurs manquantes est : 263


## Supprimez les valeurs manquantes

In [ ]:
data1=data1.dropna()

## Groupez les données par sexe (en moyenne)

In [ ]:
data1.groupby(['sex']).mean()

pclass  survived        age
sex                                  
female  2.048969  0.752577  28.687071
male    2.300912  0.205167  30.585233

## Quelle est la probabilité d'avoir survecu d'un homme et celle d'une femme ?

En observant le tableau précédent, on peut savoir que :

- La probabilité d'avoir survecu d'un homme est : 0.205167


- La probabilité d'avoir survecu d'une femme est : 0.752577

## Quel est l'age moyen des hommes et celui des femmes

En observant le tableau précédent, on peut savoir que :


- L'age moyen des hommes est : 30.585233


- L'age moyen des femmes est : 28.687071

## Groupez les données par sex et pclass

In [ ]:
data1.groupby(['sex','pclass']).mean()

survived        age
sex    pclass                     
female 1       0.962406  37.037594
       2       0.893204  27.499191
       3       0.473684  22.185307
male   1       0.350993  41.029250
       2       0.145570  30.815401
       3       0.169054  25.962273

### - Quelle est la probabilité d'avoir survecu d'un homme en troisième classe et d'une femme en première classe ?

En observant le tableau précédent, on peut savoir que :

- La probabilité d'avoir survecu d'un homme en troisième classe est : 0.169054


- La probabilité d'avoir survecu d'une femme en première classe est : 0.962406

## Convertir le champs 'sex' en données numériques

In [ ]:
data1['sex'].replace(['female','male'],[0,1],inplace=True)
data1

pclass  survived  sex  \
name                                                                     
Allen, Miss. Elisabeth Walton                         1         1    0   
Allison, Master. Hudson Trevor                        1         1    1   
Allison, Miss. Helen Loraine                          1         0    0   
Allison, Mr. Hudson Joshua Creighton                  1         0    1   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)       1         0    0   
...                                                 ...       ...  ...   
Youseff, Mr. Gerious                                  3         0    1   
Zabour, Miss. Hileni                                  3         0    0   
Zakarian, Mr. Mapriededer                             3         0    1   
Zakarian, Mr. Ortin                                   3         0    1   
Zimmerman, Mr. Leo                                    3         0    1   

                                                     age  
name                                                      
Allen, Miss. Elisabeth Walton                    29.0000  
Allison, Master. Hudson Trevor                    0.9167  
Allison, Miss. Helen Loraine                      2.0000  
Allison, Mr. Hudson Joshua Creighton             30.0000  
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  25.0000  
...                                                  ...  
Youseff, Mr. Gerious                             45.5000  
Zabour, Miss. Hileni                             14.5000  
Zakarian, Mr. Mapriededer                        26.5000  
Zakarian, Mr. Ortin                              27.0000  
Zimmerman, Mr. Leo                               29.0000  

[1046 rows x 4 columns]